### PR TITLE
Feature shopping cart

### PR DESCRIPTION
**Pull request by pvfermat**
Shopping cart functionality

Features:
- [x] Automatically create an instance of a Cart that will persist across sessions.
- [x] Retrieve all cart items
- [x] Incr./Decr. cart item
- [x] Delete cart item

### CODE REVIEW

**Comment by pvfermat** [comment]
Updated shopping cart implementation to use the `TokenValidator` and place cart within the user path of the api.

File: api/routes/cart/cartRouter.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  })
+  .catch((err) => {
+	body.msg = `Failed to remove book ${isbn}`;
+	res.status(400).send(body);
+  });
+});
+
+function updateCartItem(req, res, body, isbn, quantity) {
+  db
+  .ref(`users/Cart/${isbn}`)
```
Comment (augustoca21) [inline api/routes/cart/cartRouter.js:115]: Wouldn't be something like `users/${req.uid}/cart/${isbn}` ?

Each user should have a different cart.

File: api/routes/routesController.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ const router = express.Router() ;
 
 router.use("/books", require("./books/booksRouter.js")) ;
 router.use("/users", require("./users/usersRouter.js")) ;
+router.use("/users/cart", require("./cart/cartRouter.js"));
```
Comment (augustoca21) [inline api/routes/routesController.js:6]: Good idea, although it is related to the user, it is separated so the user file doesn't overload verbosely.

**Review by augustoca21 — APPROVED** [review]

File: api/routes/cart/cartRouter.js (hunk 1 above)
Comment (pvfermat) [reply api/routes/cart/cartRouter.js]: You're right, I reverted the changes.

File: api/routes/routesController.js (hunk 2 above)
Comment (pvfermat) [reply api/routes/routesController.js:6]: Yeah that's what I thought too. I ran my tests and it did not affect anything negatively that I could see.

**Pull request merged**